### ISSUE TITLE
`anyPattern` of type `apiextensions.JSON` breaks for kyverno cluster policy CRD in k8s v1.23 but works for previous k8s versions

### ISSUE THREAD

**Issue by mritunjaysharma394** [kubernetes/kubernetes#107690]
### What happened?

I have been working on the issue (https://github.com/kyverno/kyverno/issues/3004) and the problem that we are facing is that, we have `anyPattern` defined in [api/kyverno/v1/policy_types.go](https://github.com/kyverno/kyverno/blob/5ad0d152402f690bd5af2a14ce12d1608ca7d4ca/api/kyverno/v1/policy_types.go#L483) as: 
```AnyPattern apiextensions.JSON `json:"anyPattern,omitempty" yaml:"anyPattern,omitempty"```

While this gets parsed and works as expected in k8s `1.21` and `1.22` but it somehow breaks in `v1.23` and what we get rather as is an empty `anyPattern` after we check the applied cluster policy using:

`kubectl get copl -o yaml`
 
The empty anyPattern for v1.23 looks something like this:
 ```     
 anyPattern:
      - {}
      - {}
 ```

### What did you expect to happen?

Which should have been otherwise something like this (and it comes like this for k8s v1.21 and v1.22 but not for 1.23):
```
      anyPattern:
      - spec:
          =(ephemeralContainers):
          - =(securityContext):
              =(runAsNonRoot): true
          =(initContainers):
          - =(securityContext):
              =(runAsNonRoot): true
          containers:
          - =(securityContext):
              =(runAsNonRoot): true
          securityContext:
            runAsNonRoot: true
  ```

### How can we reproduce it (as minimally and precisely as possible)?

By applying this Kyerno cluster policy CRD  fo k8s v1.23: https://github.com/kyverno/kyverno/blob/main/config/crds/kyverno.io_clusterpolicies.yaml

### Kubernetes version

<details>

```console
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.1", GitCommit:"86ec240af8cbd1b60bcc4c03c20da9b98005b92e", GitTreeState:"clean", BuildDate:"2021-12-16T11:41:01Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.0", GitCommit:"ab69524f795c42094a6630298ff53f3c3ebab7f4", GitTreeState:"clean", BuildDate:"2021-12-08T00:29:07Z", GoVersion:"go1.17.3", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
local kind cluster
</details>

### OS version

<details>

```console
On Linux:
$ cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.3 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.3 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
$ uname -a
Linux mritunjay 5.13.0-27-generic #29~20.04.1-Ubuntu SMP Fri Jan 14 00:32:30 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
```
</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by justaugustus** [comment]
@kubernetes/release-engineering -- Let's keep an eye on this regression!
ref: https://kubernetes.slack.com/archives/CJH2GBF7Y/p1642794167092000

**Comment by MadhavJivrajani** [comment]
@liggitt is this good to close considering the cherry-pick is closed?

**Comment by liggitt** [comment]
yes, fix is merged to 1.23 release branch and will be included in 1.23.3

**Comment by liggitt** [comment]
1.23.3 has been released with this fix
